presto-main/src/main/java/io/prestosql/sql/DynamicFilters.java

```diff
@@ -129,15 +133,22 @@ public ExtractResult(List<Expression> staticConjuncts, List<Descriptor> dynamicC
 
     public static final class Descriptor
     {
+        private final String sourceId;
```

Comment:
What's the rationale for additional parameter? `id` should be unique globally and it should be possible to map it to plan node in coordinator.
Also join plan node id could possibly change while dynamic filters `ids` don't, see: `RemoveUnsupportedDynamicFilters`

I would not rely on `node id` in dynamic filters. They have unspecified semantics.

Reply:
I've now changed the code to avoid adding join node id in dynamic filter and get the dynamic filter to join node mapping by traversing the plan in DistributedExecutionPlanner.